ext/curl/multi.c

```diff
@@ -187,7 +187,12 @@ PHP_FUNCTION(curl_multi_select)
 
 	mh = Z_CURL_MULTI_P(z_mh);
 
-	error = curl_multi_wait(mh->multi, NULL, 0, (unsigned long) (timeout * 1000.0), &numfds);
+	if (!(timeout >= ((double)INT_MIN / 1000.0) && timeout <= ((double)INT_MAX / 1000.0))) {
```

Comment:
I don't think it generally makes sense to support negative values here, and given that previously we casted to unsigned, there is no need to support these.

And it may or not may be a good idea to add some general range detection to zend_range_check.h and reuse it here.

Reply:
in fact curl_multi_wait itself does not accept negative it was more I let "curl takes care of business". But I can remove the negative part.

Reply:
Ah, I didn't know this! Still it seems to me that we should catch that with a nice error message, instead of having developers deal with `curl_error()` and friends.